### DISCUSSION TITLE
[toString]

### DISCUSSION THREAD

**Discussion by wikiwiki000** [abrahamjuliot/creepjs#151] [Q&A] [answered]
---

**Comment by abrahamjuliot** [comment, marked as answer]
That is a great question. In short, if we call the `toString` method on each of these (`getDate`, etc... methods), the result should be `'function getDate() { [native code] }'` or `'function getDate() {\n    [native code]\n}'` depending on the JS engine. When object methods like this are overwritten or spoofed, the `toString` method no longer returns the native result. Instead, the result leaks the tampering code. For example, open the console in your browser and enter this with the extension on:

```js
Date.prototype.setDate.toString()
```

```js
Date.prototype.setDate.toString().toString()
```

Here's an in dept article on this topic.

https://adtechmadness.wordpress.com/2019/03/23/javascript-tampering-detection-and-stealth/
